### ISSUE TITLE
Hide room use by verifying

### ISSUE THREAD

**Issue by Biep** [vector-im/element-web#15574]
Currently, when verifying someone, one gets an invitation to a room.  This is unwanted, for many reasons:
- Most people will defer the invitation, as they are busy verifying - and then the verification won't proceed.
- Accepting the invitation leaves one with an unwanted room, that one needs to leave again afterwards.  It is messy, both the procedure and the resulting litter.
- It is quite possible one doesn't want a private channel with the person one is verifying, so it is an invasion of privacy.  The verification may be for the sake of a large group, and one might even have told Element to ignore invitations from that particular room member.  In some settings (reconciliation, e.g. of a victim with a bully or abuser) it is positively unacceptable; in others (e.g. young girl with older male schoolmate) very unadvisable.
- In the case of minors or vulnerable people, parents or guardians may not want unsupervised private channels with unvetted people - especially not if their pupil and the stranger both have full power..
- Verification can even be abused to get a private channel: if the older boy can get the young girl to believe that verification makes things more safe, she'll be an easy victim.
- Often there is already a private room for the two members, and confusion will result, leading to discussions being arbitrarily distributed over two channels.
- It scares newcomers off. (Oh, this is some complicated system - first I need to verify, and now I need to accept an invitation while the verification isn't even done yet!)

Using a room under the hood is fine - that comes with the idea that everything is a room.  But please don't leak that to the user by default.

**Comment by jryans** [comment]
Hmm, interesting feedback... The in-room verification was imagined as a good way to keep a log of your verification activity with someone without needing to invent yet another view of actions and history that is somehow separate from rooms.

In some of these cases, the room itself seems like the least of your worries though: to do verification as intended, you should be comparing things OOB, such as emoji in person (more likely on a video call or similar), so the current system somewhat assumes and expects that you would be verifying people already know personally. 

It's not really expected for you to verify random people "just because"... (I do not doubt it happens, I am just trying to add context of what was imagined when designing it.)

This is would be a significant change to the verification system, so I think we'd need to process a lot more feedback before we're ready to change something here. In any case, I'll pass this along to the team.

**Comment by philip-rehorst** [comment]
Maybe this should be an issue by itself but I see some connection from a user-perspective:

When the option _Never send encrypted messages to unverified sessions from this session_ is selected, verification seems to break. While the option is selected the Process couldn't be started, I only saw the spinning wheel for some time. I could only cancel the request.
Once the Option was disabled, verification was possible.

![2020-10-27_14h22_53](https://user-images.githubusercontent.com/62654817/97307876-87d8ee00-1860-11eb-8964-942480b500c4.png)

**Comment by jryans** [comment]
@philip-rehorst Yes, that's something entirely separate. We have https://github.com/vector-im/element-web/issues/13656 for this. In general, please default to filing a new issue if you are unsure.

**Comment by Biep** [comment]
I don't think a _significant_ change is needed - just make the room throw-away, skip the need for explicitly accepting the invitation, and make it all happen behind the scenes.
(Maybe some of those **do** require significant changes, but from a user point of view they don't seem particularly invasive.)

**Comment by Biep** [comment]
Maybe it is good to separate out some issues.

- The privacy issues will be mostly solved if the room has no write rights for its members.
- The complexity issues will be mostly solved if the room is dealt with behind the screens - Element doing the acceptation of the invitation, and not showing the room by default.
- Either part will mostly solve the confusion issue (conversations spreading over several rooms).

**Comment by mureni** [comment]
I wholeheartedly agree with the others mentioned, and have often grumbled about this UX flow with others for many of the same reasons. Specifically, even in a tech-friendly and privacy-focused group, opening what is to all appearances a new private message window causes massive confusion. The idea that "I don't want to accept this invite" or "I don't want to switch my current view" because you are waiting for the verification to complete is spot on. It is also exacerbated by the fact that if you DO switch windows, often the "awaiting verification" panel will disappear until you actively select the user pending verification again.  This can lead to confusion about whether the process failed, or if it worked, and removes the obvious next step interaction ("yes, they match" in the case of emoji verification) from the view. 

If opening a new room is necessary, I would strongly suggest some visual indicator to show that this is not a direct/private message, it is a verification process -- and as Biep mentioned above, ensure the members have no write power in the verification room. This would also prevent screencapping the emoji verification within the room itself, which I have seen frequently as a quick and dirty (and insecure) way to get the verification out of the way.
